### PR TITLE
Built scripts are exported as consumable build artifact

### PR DESCRIPTION
**Pull request by erichaagdev**
This PR extends the build so that the scripts are exported as a consumable build artifact. This makes it so that the built scripts can be included and referenced in a composite build.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -9,6 +11,8 @@ plugins {
     id("org.gradle.wrapper-upgrade") version "0.11.1"
 }
 
+group = "org.gradle"
```
Comment (etiennestuder) [inline build.gradle.kts:14]: Please use `com.gradle`, like we do for CCUD, etc. (given this is GE related).

File: build.gradle.kts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@file:Suppress("HasPlatformType")
```
Comment (etiennestuder) [inline build.gradle.kts:1]: Do we need this and if so, why?

File: build.gradle.kts [hunk 3]
```diff
@@ -171,6 +175,23 @@ tasks.register<Task>("copyScripts") {
     dependsOn(copyGradleScripts, copyMavenScripts)
 }
 
+val consumableGradleScripts by configurations.creating {
+    isCanBeConsumed = true
+    isCanBeResolved = false
+    attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named("gradle-build-validation-scripts"))
```
Comment (etiennestuder) [inline build.gradle.kts:181]: I'm not familiar with the Usage.USAGE_ATTRIBUTE attribute? Can you add a pointer to the review, just for my education, please?

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 3 above)
Comment (erichaagdev) [reply build.gradle.kts:181]: [Here are the docs](https://docs.gradle.org/current/userguide/variant_attributes.html#ecosystem_independent_standard_attributes) and [here is the Javadoc](https://docs.gradle.org/current/javadoc/org/gradle/api/attributes/Usage.html)..

File: build.gradle.kts (hunk 2 above)
Comment (erichaagdev) [reply build.gradle.kts:1]: It is to suppress these warnings:

image

Alternatively, we could specify the type explicitly. For example:

```kotlin
val consumableGradleScripts: Configuration by configurations.creating {
  ...
}
```

File: build.gradle.kts (hunk 1 above)
Comment (erichaagdev) [reply build.gradle.kts]: Done in 20e1a85d9309aa7e2a9028d1dd53c7246611bb90

File: build.gradle.kts (hunk 2 above)
Comment (etiennestuder) [reply build.gradle.kts:1]: Thanks for the insights.

File: build.gradle.kts (hunk 3 above)
Comment (etiennestuder) [reply build.gradle.kts:181]: Thanks.

**Review by etiennestuder — APPROVED** [review]

**Pull request merged**